src/MongoDB/Driver/Cursor.cpp

```diff
@@ -71,12 +57,39 @@ Object hippo_cursor_init(mongoc_cursor_t *cursor, mongoc_client_t *client)
 	cursor_data->client = client;
 	cursor_data->m_server_id = mongoc_cursor_get_hint(cursor);
 	cursor_data->next_after_rewind = 0;
+	cursor_data->m_read_preference = readPreference;
+	cursor_data->m_db = NULL;
+	cursor_data->m_collection = NULL;
+	cursor_data->current = -1;
 
 	hippo_cursor_load_current(cursor_data);
 
 	return obj;
 }
 
+Object hippo_cursor_init_for_command(mongoc_cursor_t *cursor, mongoc_client_t *client, const char *db, const Variant &command, const Variant &readPreference)
```

Comment:
Why `const Variant &command` here? It seems inconsistent with the query version, and you use an object in the following function, which calls this:

```HPHP::Object Utils::doExecuteCommand(
    const char *db,
    mongoc_client_t *client,
    int server_id,
    const HPHP::Object &command,
    const HPHP::Variant &readPreference)
```

Am I correct in assuming variant is only needed when the argument may be an object or null?